src/components/Charts/LineChart.tsx

```diff
@@ -0,0 +1,50 @@
+import { Group } from '@visx/group'
+import { LinePath } from '@visx/shape'
+import { CurveFactory } from 'd3'
+import { radius } from 'd3-curve-circlecorners'
+import { ReactNode } from 'react'
+import { useTheme } from 'styled-components/macro'
+import { Color } from 'theme/styled'
+
+interface LineChartProps<T> {
+  data: T[]
+  getX: (t: T) => number
+  getY: (t: T) => number
+  marginTop: number
+  curve?: CurveFactory
+  color?: Color
+  strokeWidth: number
+  children?: ReactNode
+  width: number
+  height: number
+}
+
+export default function LineChart<T>({
```

Comment:
consider memoizing this component since it may be expensive to render: https://reactjs.org/docs/react-api.html#reactmemo